File: app/src/main/java/com/ivanmagda/habito/models/ReminderTime.java
package com.ivanmagda.habito.models;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

public final class ReminderTime {

    private int mHour;
    private int mMinutes;

    private static final SimpleDateFormat FORMATTER = new SimpleDateFormat("HH:mm", Locale.getDefault());

    public ReminderTime() {
        final Calendar c = Calendar.getInstance();
        this.mHour = c.get(Calendar.HOUR_OF_DAY);
        this.mMinutes = c.get(Calendar.MINUTE);
    }

    public ReminderTime(int hour, int minutes) {
        this.mHour = hour;
        this.mMinutes = minutes;
    }

    public int getHour() {
        return mHour;
    }

    public void setHour(int hour) {
        this.mHour = hour;
    }

    public int getMinutes() {
        return mMinutes;
    }

    public void setMinutes(int minutes) {
        this.mMinutes = minutes;
    }

    public String getTimeString() {
        return FORMATTER.format(getTodayReminderDate());
    }

    public static String getTimeString(int hour, int minutes) {
        return getTimeString(new ReminderTime(hour, minutes));
    }

    public static String getTimeString(ReminderTime reminderTime) {
        return reminderTime.getTimeString();
    }

    public Date getTodayReminderDate() {
        Calendar calendar = Calendar.getInstance();
        calendar.set(Calendar.HOUR_OF_DAY, mHour);
        calendar.set(Calendar.MINUTE, mMinutes);
        calendar.set(Calendar.SECOND, 0);
        calendar.set(Calendar.MILLISECOND, 0);
        return calendar.getTime();
    }

    @Override
    public String toString() {
        return "ReminderTime{" +
                "mHour=" + mHour +
                ", mMinutes=" + mMinutes +
                '}';
    }

}

File: app/src/main/java/com/ivanmagda/habito/models/ReminderTime.kt
package com.ivanmagda.habito.models

import java.text.SimpleDateFormat
import java.util.*

class ReminderTime(val hour: Int, val minutes: Int) {

    val timeString: String
        get() = FORMATTER.format(todayReminderDate)

    val todayReminderDate: Date
        get() {
            val calendar = Calendar.getInstance()
            calendar.set(Calendar.HOUR_OF_DAY, hour)
            calendar.set(Calendar.MINUTE, minutes)
            calendar.set(Calendar.SECOND, 0)
            calendar.set(Calendar.MILLISECOND, 0)

            return calendar.time
        }

    override fun toString(): String {
        return "ReminderTime{" +
                "mHour=" + hour +
                ", mMinutes=" + minutes +
                '}'
    }

    companion object {

        private val FORMATTER = SimpleDateFormat("HH:mm", Locale.getDefault())

        fun getTimeString(hour: Int, minutes: Int): String {
            return getTimeString(ReminderTime(hour, minutes))
        }

        fun getTimeString(reminderTime: ReminderTime): String {
            return reminderTime.timeString
        }
    }
}
